### ISSUE TITLE
Add a time variable in NetCDF output file

### ISSUE THREAD

**Issue by wietsefranssen** [UW-Hydro/VIC#297]

**Comment by wietsefranssen** [comment]
Together with the time variable we also want to include the lat and lon variables in the NetCDF output.
We want to do this as much ad possible following the cf-conventions.

We prefer have one dimensional lat/lon variables like the standard way:
see chapter 5.1: http://cfconventions.org/Data/cf-conventions/cf-conventions-1.6/build/cf-conventions.html#coordinate-system
however chapter 5.2 is also possible but less used and also less supported by our analysis software packages.

What is your opinion about this?

**Comment by jhamman** [comment]
Yes, we want to be able to include the x (lon) and y (lat) coordinates in the output file.  We do however, need to be able to support 1 and 2 dimensional coordinates.  So the answer I think is both.

**Comment by wietsefranssen** [comment]
the standard one dimensional way is:
        double lon(lon) ;
                lon:long_name = "longitude" ;
                lon:units = "degrees_east" ;
                lon:standard_name = "longitude" ;
        double lat(lat) ;
                lat:long_name = "latitude" ;
                lat:units = "degrees_north" ;
                lat:standard_name = "latitude" ;

the 2-dim ways is: 
float xc(xc) ;
xc:axis = "X" ;
xc:long_name = "x-coordinate in Cartesian system" ;
xc:units = "m" ;
float yc(yc) ;
yc:axis = "Y" ;
yc:long_name = "y-coordinate in Cartesian system" ;
float lon(yc,xc) ;
lon:long_name = "longitude" ;
lon:units = "degrees_east" ;
float lat(yc,xc) ;
lat:long_name = "latitude" ;
lat:units = "degrees_north" ;

this means that if we want to follow the protocol we only can choice between one of the two options the same time (naming conflict). In order to solve this we have three options:
1. rename the lon and lat var names in something else
2. user can choose in vic param file (or somewhere else?) which one to use.
3. VIC detects if the grid is rectangular or not (in first case it outputs the 1D ways, else it write the 2D way)

**Comment by wietsefranssen** [comment]
An other thing (FYI): we need the lat lons of the full domain (no NANs because of the mask).
Thats why I includes the following lines in get_global_domain.c

    // if MASTER_PROC
    global_domain->locations_grid = (location_grid_struct *)
                               malloc(global_domain->n_ny * global_domain->n_nx *
                                      sizeof(location_grid_struct));
    if (global_domain->locations_grid == NULL) {
        log_err("Memory allocation error in get_global_domain().");
    }
    for (i = 0; i < global_domain->n_ny * global_domain->n_nx; i++) {
        initialize_location_grid(&(global_domain->locations_grid[i]));
    }

**Comment by jhamman** [comment]
I think it is fine to have the user choose the type of grid they are using.

I'm not sure I'm following your last comment...?

**Comment by wietsefranssen** [comment]
Okay we let the user decide (any suggestion about there to add a flag (global config file/a header file?)?

**Comment by wietsefranssen** [comment]
in  get_global_domain.c VIC reads the lats and lons from the domain file (currently Stehekin_image_test.domain.nc). it stores only the lats and lons of the cells which are runned by vic (in this case 16 lat/lons). We need all lat/lons (x*y=20).

This is why I store all the lat/lons in an extra struct (location_grid_struct).

**Comment by jhamman** [comment]
Global configuration file.  We want to add the following options (names were chosen off the top of my head and @bartnijssen may suggest something different):

```
# Domain file info
COORD_DIMS 2
X_VAR_NAME lon
Y_VAR_NAME lat
MASK_VAR_NAME mask
AREA_VAR_NAME area
```

**Comment by wietsefranssen** [comment]
perfect,
I will add the first (maybe also the rest) of the variables in de global config file.

**Comment by bartnijssen** [comment]
@wietsefranssen : Why do you need the lat - lots for the whole domain (including the grid cells for which VIC is not run)? To define the output grid?

**Comment by wietsefranssen** [comment]
1: For the standard software we use to analyse VIC results we need all lat/lons of the gridcells.

2: In case you use 1D lat/lon variables (regulair grid) it is also needed to get the starting and ending positions of the lat/lons. In case many gridcells are masked out, this information is not always available.

**Comment by wietsefranssen** [comment]
can be closed (see #303).
